src/bin/main.rs

```diff
@@ -51,7 +53,18 @@ pub struct Args {
 
 fn run (args: Args) -> Result<(), Error> {
     let file = File::create(Path::new(&args.filename))?;
-    let mut obj = ArtifactBuilder::new(Target::X86_64)
+    let target = Triple {
```

Comment:
This is awesome; no matter the decision something like this is needed eventually and I think it’s appropriate for it to be a separate crate